Sources/Rendering/Core/ImageSlice/index.js

```diff
@@ -88,19 +89,11 @@ function vtkImageSlice(publicAPI, model) {
     ) {
       vtkDebugMacro('Recomputing bounds...');
       model.mapperBounds = bds.map((x) => x);
-      const bbox = [
-        vec3.fromValues(bds[1], bds[3], bds[5]),
-        vec3.fromValues(bds[1], bds[2], bds[5]),
-        vec3.fromValues(bds[0], bds[2], bds[5]),
-        vec3.fromValues(bds[0], bds[3], bds[5]),
-        vec3.fromValues(bds[1], bds[3], bds[4]),
-        vec3.fromValues(bds[1], bds[2], bds[4]),
-        vec3.fromValues(bds[0], bds[2], bds[4]),
-        vec3.fromValues(bds[0], bds[3], bds[4]),
-      ];
+      const bbox = [];
+      vtkBoundingBox.getCorners(bds, bbox);
```

Comment:
it would seem that it would be easy to change the API of `vtkBoundingBox.getCorners` to allow the following:
```
const bbox = vtkBoundingBox.getCorners(bds);
```

Reply:
Yes, but I don't want to lose focus on the topic here which was removing the use of vec3.create() as it was using Float32. We are drifting a bit. Personally I think it should have an additional signature that takes a Float64Array(24) as the current version requires 9 heap allocs. And generally I think it is best to pass in the return argument that way we can potentially avoid one heap alloc if the same memory is used every time (not the case above but it could be). 

But again, the goal of this MR was to remove vec3.create(); and friends due to their use of Float32. Personally I think many of the matrix allocs in ImageReslice should be class conts, there is no reason to realloc them every time. That would be an easy change, but the ImageReslice code is using Float64 and so for the purposes of this MR it is fine. And there are a lot of similar cases I saw throughout the code. Heap allocs === garbage collection === slow rendering and we'll have to keep our eye on them and clean them up where we can. Just looking at the diffs this topic should remove a ton of heap allocs, upgrade a lot of calcs to double, and add a few heap allocs due to the now required conversion back to Float32 in some cases for WebGL. But the use of double is the right move throughout vtk.js.

Reply:
I Agree, it might not be the focus of this MR.

I did not mean to remove the output as parameter but make it optional (slight decrease in perfs though) and return the output:  `export function getCorners(bounds, corners=[]) {... return corners;}`

I like your `Float64Array(24)` suggestion.